central/notifications/datastore/internal/writer/writer_impl.go

```diff
@@ -0,0 +1,139 @@
+package writer
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/notifications/datastore/internal/store"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var (
+	_ Writer = (*writerImpl)(nil)
+
+	rootNamespaceUUID = uuid.Nil
+)
+
+type writerImpl struct {
+	mutex sync.Mutex
+
+	buffer map[string]*storage.Notification
+	store  store.Store
+}
+
+func (c *writerImpl) read(id string) (*storage.Notification, bool) {
```

Comment:
Typically when something like `read` and `write` is performed _without_ acquiring a lock and it's expected that this done _outside_ of the function, we add the suffix `NoLock`. The group datastore has [a couple of examples for that](https://github.com/stackrox/stackrox/blob/43f83925d2fb7bd1b9d6c27a6fd733beb6743377/central/group/datastore/datastore_impl.go#L215).